prompting/tools/datasets/base.py

```diff
@@ -47,9 +47,9 @@ def get(self, name):
 
     def next(
         self, method: str = "random", selector: Selector = Selector(), **kwargs
-    ) -> Dict:
-        tries = 1
-        t0 = time.time()
+    )->Context:
+        tries : int = 1
+        t0 : float = time.time()
```

Comment:
we don't need type hints for variables defined inplace

```suggestion
        tries = 1
        t0 = time.time()
```